AsyncDisplayKit/ASMapNode.mm

```diff
@@ -159,6 +161,15 @@ - (void)setRegion:(MKCoordinateRegion)region
 
 - (void)takeSnapshot
 {
+  // do not trigger snapshot before layout
+  ASLayout *layout = self.calculatedLayout;
+  if (layout == nil || CGSizeEqualToSize(CGSizeZero, layout.size)) {
```

Comment:
@hannahmbanana this is a bit confusing to read. If my understanding is correct, it'd be nice to have a comment which says, set _snapshotNeeded to YES so if layout changes in the future, we'll try snapshotting again.